src/modules/Login/action.js

```diff
@@ -32,6 +33,61 @@ import type { Dispatch, GetState } from '../ReduxTypes'
 
 const localeInfo = Locale.constants() // should likely be moved to login system and inserted into Redux
 
+const createDefaultWallets = async (account: EdgeAccount, defaultFiat: string) => {
+  const ethWalletName = s.strings.string_first_ethereum_wallet_name
+  const btcWalletName = s.strings.string_first_bitcoin_wallet_name
+  const bchWalletName = s.strings.string_first_bitcoincash_wallet_name
+  const ethWalletType = Constants.ETHEREUM_WALLET
+  const btcWalletType = Constants.BITCOIN_WALLET
+  const bchWalletType = Constants.BITCOINCASH_WALLET
+  const fiatCurrencyCode = 'iso:' + defaultFiat
+
+  let edgeWallet
+  if (global.currencyCode) {
+    let walletType, walletName
+    // We got installed via a currencyCode referral. Only create one wallet of that type
+    for (const pluginName in account.currencyConfig) {
+      const { currencyInfo } = account.currencyConfig[pluginName]
+      if (currencyInfo.currencyCode.toLowerCase() === global.currencyCode.toLowerCase()) {
+        walletType = currencyInfo.walletType
+        walletName = sprintf(s.strings.my_crypto_wallet_name, currencyInfo.displayName)
+        global.startMoment && global.startMoment('INIT_ACCOUNT_CREATE_ONE_WALLET')
+        try {
+          edgeWallet = await runWithTimeout(20000, account.createCurrencyWallet(walletType, { name: walletName, fiatCurrencyCode }))
+        } catch (e) {
+          if (e.name === 'TimeoutExceeded') {
+            e.message = s.strings.error_creating_wallets
+          }
+          throw e
+        }
+        global.endMoment && global.endMoment('INIT_ACCOUNT_CREATE_ONE_WALLET')
+      }
+    }
+  }
+  if (!edgeWallet) {
+    global.startMoment && global.startMoment('INIT_ACCOUNT_CREATE_WALLETS')
+    try {
+      edgeWallet = await runWithTimeout(20000, account.createCurrencyWallet(btcWalletType, { name: btcWalletName, fiatCurrencyCode }))
+      await runWithTimeout(20000, account.createCurrencyWallet(bchWalletType, { name: bchWalletName, fiatCurrencyCode }))
+      await runWithTimeout(20000, account.createCurrencyWallet(ethWalletType, { name: ethWalletName, fiatCurrencyCode }))
+      // const p = []
+      // p.push(account.createCurrencyWallet(btcWalletType, { name: btcWalletName, fiatCurrencyCode }))
+      // p.push(account.createCurrencyWallet(bchWalletType, { name: bchWalletName, fiatCurrencyCode }))
+      // p.push(account.createCurrencyWallet(ethWalletType, { name: ethWalletName, fiatCurrencyCode }))
+      // const results = await runWithTimeout(20000, Promise.all(p))
+      // edgeWallet = results[0]
```

Comment:
This is all commented out because core errors if we make multiple wallets at once.